### PR TITLE
ARROW-10633: [Rust][DataFusion] Dependency version updates

### PR DESCRIPTION
**Pull request by Dandandan**
Upgrades versions to latest one. Brings in tokio 0.3, crossbeam 0.8 and some smaller updates.

### CODE REVIEW

**Review by nevi-me — APPROVED** [review]

File: rust/datafusion/Cargo.toml [hunk 1]
```diff
@@ -50,14 +50,14 @@ parquet = { path = "../parquet", version = "3.0.0-SNAPSHOT", features = ["arrow"
 sqlparser = "0.6.1"
 clap = "2.33"
 rustyline = {version = "6.0", optional = true}
-crossbeam = "0.7"
-paste = "0.1"
+crossbeam = "0.8"
+paste = "^1.0"
```
Comment (alamb) [inline rust/datafusion/Cargo.toml:54]: Any reason to add `^` dependency as opposed to just paste = "1.0"?

**Review by alamb — APPROVED** [review]
I also merged this into master and rand the tests locally and everything worked for me just to make sure there were no merge conflicts. There weren't!

**Comment by alamb** [comment]
Thank you @Dandandan !

**Comment by rdettai** [comment]
@alamb @Dandandan Upgrading Tokio to 0.3 is not as simple as it seams. The runtime is not compatible with the 0.2 one, and it requires to add a compatibility adapter that runs an extra thread (https://docs.rs/tokio-compat-02/0.1.2/tokio_compat_02/). 

The problem is that many libraries are still running 0.2 (as 0.3 is fairly recent):
- tonic: flight is mostly broken because of that
- rustoto: not directly used but used by external users (and not just me I guess 😄)

What was the reason for upgrading Tokio? If there is no clear reason, I would be in favor of downgrading back to 0.2.

**Comment by Dandandan** [comment]
@rdettai 
The reason was more to upgrade to the latest version to benefit from the changes and to make future upgrades easier (tokio 1.0 is also coming).

For me it would be no problem to downgrade to tokio 0.2 if it causes too much problems elsewhere and wait until other libraries catch up.

**Comment by alamb** [comment]
@rdettai  -- I am also fine downgrading back to tokio 3.0 -- I read some of the discussion on https://github.com/apache/arrow/pull/8734 and it seems like arrow flight is ok? 

Can you be clearer about what exactly is breaking for you? Is it downstream dependencies that can't be to get updated to tokio 3.0?

**Comment by rdettai** [comment]
Maybe I'm doing something wrong, but if I run:
`cd rust/arrow-flight; cargo run --example server`
on the current master, the server panics. Is it working for you?

 I can try to make the example into a test to have this verified automatically, but I wont have time to do it before the end of the week.

**Comment by alamb** [comment]
@rdettai  -- the same thing fails for me. I filed https://issues.apache.org/jira/browse/ARROW-10710

**Comment by Dandandan** [comment]
@rdettai rustoto just released a new version with 1.0 , tonic has an open PR for the upgrade. Any other dependencies we should look out for when upgrading to 1.0?

**Pull request closed without merging**